### PR TITLE
metrics/librato: ensure resp.body closed

### PR DESCRIPTION
**Pull request by jsvisa**

### CODE REVIEW

**Review by fjl — APPROVED** [review]

**Review by holiman — COMMENTED** [review]
LGTM, and we have a similar flaw also in `rpc/http_test.go`

```
	resp, err := http.DefaultClient.Do(request)
	if err != nil {
		t.Fatalf("request failed: %v", err)
	}
	confirmStatusCode(t, resp.StatusCode, expectedStatusCode)
}
```

Also a lot of tests calls `node/rpcstack.go` , the method `baseRpcRequest`. 
Maybe we should do something like this? 
```diff 
diff --git a/node/rpcstack_test.go b/node/rpcstack_test.go
index 4d10e61e2d..0790dddec4 100644
--- a/node/rpcstack_test.go
+++ b/node/rpcstack_test.go
@@ -320,6 +320,7 @@ func baseRpcRequest(t *testing.T, url, bodyStr string, extraHeaders ...string) *
        if err != nil {
                t.Fatal(err)
        }
+       t.Cleanup(func() { resp.Body.Close() })
        return resp
 }
 
```

**Comment by jsvisa** [comment]
Thanks for this，I will fix all of them today

File: rpc/http_test.go [hunk 1]
```diff
@@ -94,6 +95,8 @@ func confirmHTTPRequestYieldsStatusCode(t *testing.T, method, contentType, body
 	if err != nil {
 		t.Fatalf("request failed: %v", err)
 	}
+	defer resp.Body.Close()
+	io.Copy(io.Discard, resp.Body)
```
Comment (holiman) [inline rpc/http_test.go:99]: Is this really needed? Any docs saying this is needed?

File: graphql/graphql_test.go [hunk 2]
```diff
@@ -238,6 +239,7 @@ func TestGraphQLBlockSerializationEIP2718(t *testing.T) {
 		if err != nil {
 			t.Fatalf("could not post: %v", err)
 		}
+		defer resp.Body.Close()
 		bodyBytes, err := io.ReadAll(resp.Body)
```
Comment (holiman) [inline graphql/graphql_test.go:243]:
```suggestion
		bodyBytes, err := io.ReadAll(resp.Body)
		resp.Body.Close()
```

File: p2p/simulations/mocker_test.go [hunk 3]
```diff
@@ -145,15 +146,17 @@ func TestMocker(t *testing.T) {
 	if err != nil {
 		t.Fatalf("Could not stop mocker: %s", err)
 	}
+	defer resp.Body.Close()
```
Comment (holiman) [inline p2p/simulations/mocker_test.go:149]:
```suggestion
	resp.Body.Close()
```

**Review by holiman — APPROVED** [review]
LGTM

**Pull request merged**